src/Translator.php

```diff
@@ -0,0 +1,542 @@
+
+            // Commit changes
+            $title = 'ENH Update translations';
+            $this->exec("git commit -m \"$title\"", $modulePath);
+
+            if ($this->isDevMode) {
+                $this->log('Not pushing changes or creating pull-request because TX_DEV_MODE is enabled');
+                return;
+            }
+
+            // Push changes to creative-commoners
+            $this->exec("git push --set-upstream tx-ccs $branch", $modulePath);
+
+            // Create pull-request via github api
+            // https://docs.github.com/en/rest/pulls/pulls#create-a-pull-request
+            $body = 'Automated translations update generated using silverstripe/tx-translator';
+            $client = new Client();
+            $response = $client->request('POST', $endpoint, [
+                'headers' => [
+                    "Accept" => "application/vnd.github.v3+json",
+                    "Authorization" => "token {$this->githubToken}"
+                ],
+                'body' => json_encode((object) [
+                    "title" => "$title",
+                    "body" => $body,
+                    "head" => "creative-commoners:$branch",
+                    "base" => "$currentBranch"
+                ], JSON_PRETTY_PRINT + JSON_UNESCAPED_SLASHES + JSON_UNESCAPED_UNICODE)
+            ]);
+            $code = $response->getStatusCode();
+            $responseBody = (string) $response->getBody();
+            if ($code === 201) {
+                $pullRequestUrl = json_decode($responseBody)->html_url;
+                $this->log("Pull request was succesfully created at $pullRequestUrl");
+                $this->pullRequestUrls[] = $pullRequestUrl;
+            } else {
+                $this->log($responseBody);
+                throw new Exception("Pull request failed - status code was $code");
+            }
+        }
+    }
+
+    private function outputPullRequestUrls(): void
+    {
+        $this->log('');
+        $this->log('The following pull-requests were created:');
+        foreach ($this->pullRequestUrls as $pullRequestUrl) {
+            $this->log($pullRequestUrl);
+        }
+        $this->log('');
+    }
+
+    private function getModuleName(string $modulePath): string
+    {
+        return json_decode(file_get_contents("$modulePath/composer.json"))->name;
+    }
+
+    private function exec(string $command, ?string $cwd = null): string
+    {
+        $this->log("Running $command");
+        $process = Process::fromShellCommandline($command, $cwd);
+        $process->run();
+        if (!$process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+        return trim($process->getOutput());
+    }
+
+    private function log(string $message): void
+    {
+        echo "$message\n";
+    }
+
+    /**
+     * Gets the module lang dir
+     */
+    private function getLangDirectory(string $modulePath): string
```

Comment:
```suggestion
    private function getYmlLangDirectory(string $modulePath): string
```
In Silverstripe we tend to think of the yml lang as the default lang dir... but here we treat yml and json lang with equal importance.

Reply:
Updated